### ISSUE TITLE
Rule proposal: `equal-function-arguments`

### ISSUE THREAD

**Issue by silverwind** [sindresorhus/eslint-plugin-unicorn#1998]
### Description

A common source of bugs during refactoring is when functions are called with less or more arguments than they expect. Prior art includes [sonarjs/no-extra-arguments](https://github.com/SonarSource/eslint-plugin-sonarjs/blob/master/docs/rules/no-extra-arguments.md), but I would like to see a more strict version that also triggers on too few arguments.

### Fail

```js
const fn = (a, b) => a + b;
fn(1); // Expected 2 function arguments, but got 1
```

```js
const fn = (a, b) => a + b;
fn(1, 2, 3); // Expected 2 function arguments, but got 3
```

### Pass

```js
const fn = (a, b) => a + b;
fn(1, 2);
```

### Additional Info

Unlike `sonarjs/no-extra-arguments`, I'd not make a exception for `arguments` use because as I see it, `arguments` can and should be replaced with spread syntax.

**Comment by papb** [comment]
I'm surprised there is no rule that checks this already. Sounds so basic. I'd expect it to be present in the recommended preset of eslint itself.

**Comment by silverwind** [comment]
Should it have a option to allow omitting optional options objects? I think it's a common pattern:

```js
const fn = (a, b, {c} = {}) => {};
fn(1, 2) // should it be ok to omit when last argument is a options object?
```

**Comment by papb** [comment]
It should allow omitting all arguments that have a default value (`= something`).

**Comment by sindresorhus** [comment]
Accepted
